### PR TITLE
Modify Keytrade Bank PDF-Importer to support new transactions

### LINKED ISSUE #2605: DUPLICATA in Keytrade Bank PDF Import
**Describe the bug**
When I tried to import a Keytrade PDF file that has a DUPLICATA watermark on it, the parsing fail.
The issue is that the PDFParser output the following text:
```
Disponible depuis le : 26/11/2021 Téléchargé le : 29/12/2021A
Bordereau T
Compte titres: 7634568
Numéro de réf. : 7763C9846/01AAchat 15 KINEPOLIS GROUP (BE0974274061) à 46,3 EURType d' instrument: Actions
Type d'ordre: Limit (46,30 EUR) I
Validité: GTC
Ordre créé à: 26/P11/2021L10:05:13 CETDate et heure d'exécution: 26/11/2021 13:25:24 CETDate de comptabilisation: 26/11/2021Date valeur: 30/11/2021Lieu d'exécution: EURONEXT - EURONEXT BRUSS
Montant brut U - 694,50 EURD - 694,50 EURTaxe boursière - 0,35% - 2,43 EURCourtage - 7,50 EUR
Débit 704,43 EUR
Keytrade Bank, succursale belge d'Arkéa Direct Bank SA (France)
Boulevard du Souverain 100 - 1170 Bruxelles, RPM Bruxelles - TVA BE 0879 257 191
Intermédiaire en assurance autorisé à exercer en Belgique (ORIAS 07008441)
T +32 2 679 90 00 - F +32 2 679 90 01 - info@keytradebank.com - keytradebank.be 
```
(The data are modified)

We can see that the "DUPLICATA" word is added in the parsed text.
I tried to fix the issue but I couldn't. I think this is an issue with the KeytradeBankPDFExtractor that can't match the needed regex as the "DUPLICATA" removes the needed linebreaks for the regex to work.
I will try to fix it when I have more time. If you know how to fix it, feel free to do it.

**To Reproduce**
1. File -> Import -> PDF Document
2. Choose a Keytrade PDF file with a DUPLICATA watermark

**Expected behavior**
The PDF is decoded/imported as if it was no DUPLICATA watermark.

**Screenshots**
[PDF File Screenshot](https://i.imgur.com/O595CeA.jpg)

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Version 20.04.3
 - App Version: 0.56.2

### PR DESCRIPTION
**Pull request by Nirus2000**
Add new buy/sell transactions
Fixes #2605
https://forum.portfolio-performance.info/t/pdf-import-from-keytrade/19238/1
https://forum.portfolio-performance.info/t/pdf-import-from-keytrade/19238/2

Hello @dimitriwauters
here is the full PDF-Importer changes for you to see in their entirety. 
You could use this as a template for other documents in the future.

For example, once you provide the appropriate PDF debugs here as a comment or in the 
Portfolio Performance forum -> [here](https://forum.portfolio-performance.info/t/pdf-import-von-keytrade-bank/15416) <- we can add them here.
It is also no problem if you write in French or English in the forum, as long as the PDF debugs are correct.

Your pull request #2604 can be closed.

Greetings
Alex

### CODE REVIEW

**Comment by dimitriwauters** [comment]
I've posted the debug info on the forum.

I have seen that you mentioned the issue #2605 but this is not related to this PR as this is another problem.
There are two types of documents: one without the DUPLICATA watermark and one with it (when you re-download the document).
Here you are solving the issue when there is **no** DUPLICATA. 
The output text is broken when the DUPLICATA word is added:
```
Disponible depuis le : 26/11/2021 Téléchargé le : 29/12/2021A
Bordereau T
Compte titres: 7634568
Numéro de réf. : 7763C9846/01AAchat 15 KINEPOLIS GROUP (BE0974274061) à 46,3 EURType d' instrument: Actions
Type d'ordre: Limit (46,30 EUR) I
Validité: GTC
Ordre créé à: 26/P11/2021L10:05:13 CETDate et heure d'exécution: 26/11/2021 13:25:24 CETDate de comptabilisation: 26/11/2021Date valeur: 30/11/2021Lieu d'exécution: EURONEXT - EURONEXT BRUSS
Montant brut U - 694,50 EURD - 694,50 EURTaxe boursière - 0,35% - 2,43 EURCourtage - 7,50 EUR
Débit 704,43 EUR
Keytrade Bank, succursale belge d'Arkéa Direct Bank SA (France)
Boulevard du Souverain 100 - 1170 Bruxelles, RPM Bruxelles - TVA BE 0879 257 191
Intermédiaire en assurance autorisé à exercer en Belgique (ORIAS 07008441)
T +32 2 679 90 00 - F +32 2 679 90 01 - info@keytradebank.com - keytradebank.be 
```

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/KeytradeBankPDFExtractor.java [hunk 1]
```diff
@@ -211,9 +216,18 @@ private void addDividendeTransaction()
...
+                .match("^Verrechnungssteuer [\\.,\\d]+ % (?<tax>[\\.,\\d]+) (?<currency>[\\w]{3})$")
+                .assign((t, v) -> processTaxEntries(t, v, type))
+
                 // Impôt à la source 26,38 % 16,35 EUR
                 .section("tax", "currency").optional()
-                .match("^(Verrechnungssteuer|Imp.t . la source) [\\.,\\d]+ % (?<tax>[\\.,\\d]+) (?<currency>[\\w]{3})$")
+                .match("^Imp.t . la source [\\.,\\d]+ % (?<tax>[\\.,\\d]+) (?<currency>[\\w]{3})$")
+                .assign((t, v) -> processTaxEntries(t, v, type))
+
+                // Taxe boursière - 0,35% - 5,22 EUR
```
Comment (dimitriwauters) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/KeytradeBankPDFExtractor.java:228]: There is no "Taxe boursière" and "courtage" with the dividend (at least for the Belgian document).
It is replaced by "Précompte mobilier belge" like you can see [here](https://forum.portfolio-performance.info/t/pdf-import-von-keytrade-bank/15416/27?u=scrayn).

**Comment by dimitriwauters** [comment]
I tried your commit with different documents and it seems there is still an issue when the DUPLICATA is present and the order type is "MARKET", a linebreak is different...
Besides, all my other sell/buy document are correcly working :)
```
PDFBox Version: 1.8.16
-----------------------------------------
Disponible depuis le : 05/11/2020 Téléchargé le : 29/12/2021A
Bordereau T
Compte titres: 7634568
Numéro de réf. : 7763C9846/01AAchat 21 KBC (BE0003565737) à 43,85 EURType d' instrument: Actions
Type d'ordre: Market I
Validité: Day
Ordre créé à: 05/11/2020 11:25:43 CET
Date et heure d'exécution: 05/P11/2020L11:25:43 CETDate de comptabilisation: 05/11/2020Date valeur: U09/11/2020Lieu d'exécution: EquiductMontant brut - 920,85 EURD - 920,85 EURTaxe boursière - 0,35% - 3,22 EURCourtage - 7,50 EUR
Débit 931,57 EUR
Keytrade Bank, succursale belge d'Arkéa Direct Bank SA (France)
Boulevard du Souverain 100 - 1170 Bruxelles, RPM Bruxelles - TVA BE 0879 257 191
Intermédiaire en assurance autorisé à exercer en Belgique (ORIAS 07008441)
T +32 2 679 90 00 - F +32 2 679 90 01 - info@keytradebank.com - keytradebank.be 

```

**Comment by dimitriwauters** [comment]
All my files are working for sell/buy!
Thanks @Nirus2000

**Comment by Nirus2000** [comment]
Hallo @dimitriwauters
Okay, it's nice that the purchases and sales are now working.
I would make the dividends tomorrow or the day after.
Just check back in the new year and check the pull request.

Have a good new year...

**Comment by Nirus2000** [comment]
Hallo @dimitriwauters 
i think it's done, so you can check.

@buchen 
Sorry für diesen schlecht durchgeführten 
pull-request mit den ganzen Änderungen.

**Comment by dimitriwauters** [comment]
Hello @Nirus2000 
Yep, that's working fine!
Thank you and have a good new year too.

**Pull request merged**